lib/cartopy/mpl/geoaxes.py

```diff
@@ -753,6 +753,20 @@ def add_geometries(self, geoms, crs, **kwargs):
 
         """
         styler = kwargs.pop('styler', None)
+        has_LineString = False
+        try:
+            for g in iter(tuple(geoms)):
+                if isinstance(g, sgeom.LineString):
+                    has_LineString = True
+                    break
```

Comment:
Doing the check this way, if you have one LineString anywhere you'll get the new styling on all of the other geometries too. It seems like the style update should only be applied to LineString geometries themselves, not the entire collection?